### PR TITLE
amp-script: Use AMP purifier config and hooks

### PR DESCRIPTION
**Pull request by choumx**
- [x] Requires https://github.com/ampproject/worker-dom/pull/113 to be merged and released.

Changes:
- Update worker-dom to 0.1.4.
- Hook up AMP's purifier config and hooks to worker-dom's DOMPurify instance.

### CODE REVIEW

File: build-system/tasks/update-packages.js [hunk 1]
```diff
@@ -110,15 +110,13 @@ function patchRegisterElement() {
...
 function patchWorkerDom() {
-  replaceInFile(
-      'node_modules/@ampproject/worker-dom/dist/index.safe.js',
-      'node_modules/@ampproject/worker-dom/dist/index.safe.patched.js',
-      // Replace local var with an ES6 export.
-      'var MainThread=function(R){',
-      'export const MainThread=function(R){');
+  fs.copyFileSync(
+      'node_modules/@ampproject/worker-dom/dist/unminified.index.safe.mjs',
+      'node_modules/@ampproject/worker-dom/dist/unminified.index.safe.mjs.js');
```
Comment (kristoferbaxter) [inline build-system/tasks/update-packages.js]: Can likely just replace the extension here.

```suggestion
      'node_modules/@ampproject/worker-dom/dist/unminified.index.safe.js');
```

File: build-system/tasks/compile.js [hunk 2]
```diff
@@ -256,7 +256,7 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
       'node_modules/set-dom/src/**/*.js',
       'node_modules/web-animations-js/web-animations.install.js',
       'node_modules/web-activities/activity-ports.js',
-      'node_modules/@ampproject/worker-dom/dist/**/*.js',
+      'node_modules/@ampproject/worker-dom/dist/unminified.index.safe.mjs.js',
```
Comment (kristoferbaxter) [inline build-system/tasks/compile.js]: If we're changing the extension below, lets fix it up here too.

```suggestion
      'node_modules/@ampproject/worker-dom/dist/unminified.index.safe.js',
```

File: extensions/amp-script/0.1/amp-script.js [hunk 3]
```diff
@@ -15,16 +15,21 @@
...
+import {addPurifyHooks, purifyConfig} from '../../../src/purifier';
 import {
   calculateExtensionScriptUrl,
 } from '../../../src/service/extension-location';
-import {dev} from '../../../src/log';
+import {dev, user} from '../../../src/log';
 import {getMode} from '../../../src/mode';
+import {
+  sanitizer,
+  upgradeElement,
```
Comment (kristoferbaxter) [inline extensions/amp-script/0.1/amp-script.js]: I like this change a lot, makes more sense to call the `upgradeElement` method directly from the module.

File: build-system/tasks/update-packages.js (hunk 1 above)
Comment (choumx) [reply build-system/tasks/update-packages.js]: I didn't want to overwrite the existing file to avoid a source of confusion.

**Comment by choumx** [comment]
Looks like --single_pass doesn't like `*.mjs.js`. Changed to `*.mjs.patched.js.`

**Pull request merged**